cmd/ls-main.go

```diff
@@ -151,11 +148,7 @@ func mainList(ctx *cli.Context) {
 			clnt, err = newClient(targetURL)
 			fatalIf(err.Trace(targetURL), "Unable to initialize target ‘"+targetURL+"’.")
 		}
-
-		err = doList(clnt, isRecursive, isIncomplete)
-		if err != nil {
-			errorIf(err.Trace(clnt.GetURL().String()), "Unable to list target ‘"+clnt.GetURL().String()+"’.")
-			continue
-		}
+		cErr = doList(clnt, isRecursive, isIncomplete)
```

Comment:
We are in a loop here, this means cErr can be set with 'nil' in the next loop and the error will be hidden.